### ISSUE TITLE
causes OOM with remote-ssh (possibly only on very large projects)

### ISSUE THREAD

**Issue by allight** [georgewfraser/java-language-server#123]
Repro:

1) Have vscode-remote installed
2) use remote-ssh to open a remote directory with the AOSP source code tree
3) Open a random java file (for example art/test/1993-fallback-non-structural/src/art/Test1993.java)
4) wait

The remote vscode will crash with a OOM error

No configuration of classpath or anything else is required.

**Comment by georgewfraser** [comment]
This is surprising, the memory utilization should be around 370 MB, even for huge projects. Are you sure you're using java-language-server, not the [Microsoft Java extension](https://marketplace.visualstudio.com/items?itemName=redhat.java), which runs eclipse in the background?

**Comment by allight** [comment]
Yes. I specifically disabled the MS java extension because it would complain incessantly about not being able to find maven build files and didn't do any completion without them. This did give me basic BCP completions which was all I really needed (I'm mostly doing C/C++ dev and only need to use java to write small self-contained tests usually).
